### DISCUSSION TITLE
How is genomespy embedded in a react app?

### DISCUSSION THREAD

**Discussion by zhangzhen** [genome-spy/genome-spy#225] [Q&A]
I have a react app and want to add a genomics vis component to it. If possible, could you guy please tell me how to do that?

Regards,
Zhen Zhang

**Comment by tuner** [comment]
Hi Zhen

To be honest, I have very little experience with React. However, making a GenomeSpy React component should be straightforward. Use the `embed` function at a proper phase of the React component's lifecycle and the `finalize` function (in the returned EmbedResult object) to destroy it if/when the React component is removed from the DOM.

An example of embeding:
https://github.com/genome-spy/genome-spy/blob/0c6e0418a7461b40da98896bfaf5e05732384e85/packages/embed-examples/src/scaleApi.js#L87

The `finalize` function:
https://github.com/genome-spy/genome-spy/blob/0c6e0418a7461b40da98896bfaf5e05732384e85/packages/core/src/types/embedApi.d.ts#L39

Let me know if this helps you getting forward.
